### PR TITLE
Avoid unnecessary pod restart on each helm chart version

### PR DESCRIPTION
**Pull request by jkroepke**
This PR changes the the input of the config checksum which triggers the reload.

On main branch, the whole configmap/secret checksum is included as checksum input. The configmap contains labels which includes the helm chart version. Each new helm chart version forces a pod restart which is not always necessary.

By define a finer input for the pod annotation checksum, unnecessary pod restarts can be avoided.

The PR implements this for

* config
* alpha config
* secrets
* htpasswd

and could be extended to

* redis
* email
* google secret

### CODE REVIEW

**Comment by pierluigilenoci** [comment]
@jkroepke, could you please add an [annotation for ArtifactHub](https://github.com/oauth2-proxy/manifests/blob/main/helm/oauth2-proxy/Chart.yaml#L35-L41) to be approved.

File: helm/oauth2-proxy/Chart.lock [hunk 1]
```diff
@@ -1,6 +1,6 @@
 dependencies:
 - name: redis
   repository: https://charts.bitnami.com/bitnami
-  version: 18.3.2
-digest: sha256:c4d7208b098ca3c44f960f02fb0e505ac6493cf00c6539ee7bf0a603371d9959
-generated: "2023-11-13T12:24:54.470708+01:00"
+  version: 18.3.3
```
Comment (pierluigilenoci) [inline helm/oauth2-proxy/Chart.lock:4]: Please put the Redis upgrade in a separate PR.

Comment (pierluigilenoci) [reply helm/oauth2-proxy/Chart.lock:4]: We already have 1 PR for that: #179

Comment (jkroepke) [reply helm/oauth2-proxy/Chart.lock]: fixed!

**Comment by jkroepke** [comment]
> @jkroepke, could you please add an [annotation for ArtifactHub](https://github.com/oauth2-proxy/manifests/blob/main/helm/oauth2-proxy/Chart.yaml?rgh-link-date=2023-12-11T10%3A07%3A54Z#L35-L41) to be approved.

Done!

**Comment by pierluigilenoci** [comment]
@jkroepke Thank you for your contribution!

**Review by pierluigilenoci — APPROVED** [review]
